Migrate framework to Eclipse E4.37 and update Maven build plugins

bahnwaerter opened the pull request:
The changes migrate the Equinox framework with all its components from Eclipse E4.32 to E4.37. The Java version remains unchanged at version 21. In addition, the Maven build plugins are updated to their latest stable version. Architecture and operating system-specific platform targets are obsolete and are no longer needed, since the specific framework dependencies (Eclipse plugins) are now listed in the feature files of Ultimate. From now on, only the general `Ultimate.target` needs to be used as the platform target and all required framework dependencies will then be resolved automatically according to the content of the feature files.

bahnwaerter committed:
Rename target platform files to match Eclipse 4.37

bahnwaerter committed:
Migrate Eclipse 4.32 to version 4.37

bahnwaerter committed:
Update Maven build plugins

bahnwaerter committed:
Update Maven SonarCube and code coverage plugin

bahnwaerter committed:
Update used OSGI-compliant Java compiler

bahnwaerter committed:
Remove unused architecture-specific target platforms

bahnwaerter committed:
Rename target platform to remove versioning info from filename

bahnwaerter commented:
These changes also facilitate the implementation of the desired feature in #749. However, the implementation will be done in a separate pull request.

bahnwaerter commented:
The changes have already been tested manually. It is possible to build and run Ultimate based on Eclipse E4.37 under
 - [x] Linux (Arch Linux)
 - [x] macOS Ventura
 - [x] Windows 10

while using Eclipse 2025-09 as IDE.

It is also possible to build Ultimate using Maven on Linux (Arch Linux) where a Maven version greater than 3.9.9 is required for this.

schuessf commented on trunk/source/BA_SiteRepository/Ultimate.target:
```diff
@@ -0,0 +1,113 @@
+<?xml version="1.0" encoding="UTF-8" standalone="no"?>
```
Is there any reason you renamed this file? I think, this makes migration a bit less convenient and requires a small change in our wiki for the installation instructions.

bahnwaerter replied:
I did that on purpose, because it's a bad idea to include version numbers in a filename when using source code versioning. With every framework migration, we would have to rename the file and lose its versioning history. After this one-time renaming, that should no longer be the case.

schuessf commented on trunk/source/BA_SiteRepository/Ultimate.target:
```diff
@@ -0,0 +1,113 @@
+        </location>
+        <location includeAllPlatforms="true" includeConfigurePhase="true" includeMode="planner" includeSource="false" type="InstallableUnit">
+            <repository id="orbit-aggregation" location="https://download.eclipse.org/tools/orbit/simrel/orbit-aggregation/2025-09"/>
+            <unit id="org.eclipse.orbit.legacy.feature.group" version="0.0.0"/>
```
You replaced all the versions by 0.0.0. Does this mean that any version that is found in the repo is accepted and thus makes future framework updates even simpler?

bahnwaerter replied:
> [...] makes future framework updates even simpler?

You recognized that correctly. The version information for all dependencies will now be resolved automatically (according to the version constraints of the plugins). This applies to both the bundles and each individual plugin. This approach for RCP development is now set as the new default in the Eclipse IDE.

bahnwaerter committed:
Use Maven 3.9.11 in Jenkins build and test pipelines

Update the Maven version used in Jenkins to version 3.9.11. This
version upgrade is necessary because Tycho 5.0.0 requires a Maven
version greater than 3.9.9.

schuessf reviewed (approved):
So, then this looks good to me. The question is, should we merge this before SV-COMP or is this somehow "too risky"?

bahnwaerter commented:
> [...] The question is, should we merge this before SV-COMP or is this somehow "too risky"?

I would tend to merge these changes after SV-COMP. We also don't have any performance comparison yet. Also, it's unclear whether the Eclipse compiler update will cause further problems (as it was last year the case).

schuessf commented:
With this framework migration, the version of `org.eclipse.equinox.launcher` has also changed, right? If this is the case, several hardcoded version strings (e.g., in [`Ultimate.py`](https://github.com/ultimate-pa/ultimate/blob/23b7c4e0426ba1abdc7a3c014efbfda56e6fdb47/releaseScripts/default/adds/Ultimate.py#L295)) still have to be adapted.

bahnwaerter committed:
Update filename of launcher Jar in startup scripts

bahnwaerter commented:
> With this framework migration, the version of `org.eclipse.equinox.launcher` has also changed, right? [...]

That's correct, thanks for the hint. I just adjusted the filename of the launcher Jar in the startup scripts.

schuessf committed:
Use junit4 in POM

schuessf commented:
I just rebased this MR and started the nightly to check if it succeeds (required 7826bda as a small fix).
Since Eclipse 2025-12 is out now, can you check if we can directly upgrade to this version (not urgent though)?

schuessf commented:
Unfortunately, there was still some issue that lead to all tests failing, but I hopefully fixed that in ~~26d3096~~ f39e5c0.
Also, I just tried updating to 2025-123 (and cleaning up the target file): https://github.com/ultimate-pa/ultimate/tree/wip/fs/update-target
Let's see, if it is working 🙂

schuessf committed:
Bugfix: Use correct import

bahnwaerter commented:
The preparation for this pull request is now complete. @danieldietsch and @maul-esel, would you like to review the changes to ensure everything is okay before merging?

danieldietsch reviewed (approved):
Looks good to me

maul-esel commented:
I can try to take a look this evening. But if I don't manage or you want to deal with this today, I'm fine if you merge directly.
